### PR TITLE
move Domain attachments to blob storage

### PR DESCRIPTION
**Pull request by nickpell**

### CODE REVIEW

File: corehq/apps/domain/models.py [hunk 1]
```diff
@@ -997,16 +1000,19 @@ def commtrack_settings(self):
...
         if not self.has_custom_logo:
             return None
 
         return (
             self.fetch_attachment(LOGO_ATTACHMENT),
-            self['_attachments'][LOGO_ATTACHMENT]['content_type']
+            (
+                self['_attachments'][LOGO_ATTACHMENT]['content_type']
+                if self['_attachments']
+                else self.external_blobs[LOGO_ATTACHMENT].content_type
```
Comment (millerdev) [inline corehq/apps/domain/models.py]: You can use [`self.blobs`](https://github.com/dimagi/commcare-hq/blob/76b69907c78bb26b450c6fb4f3281defc3c1783a/corehq/blobs/mixin.py#L62)`[LOGO_ATTACHMENT].content_type` and remove the `if`/`else` and `_attachments` lookup.

File: corehq/blobs/migrate.py [hunk 2]
```diff
@@ -649,6 +650,7 @@ def migrate(self, filename=None, reset=False, max_retry=2, chunk_size=100, limit
 
 
 MIGRATIONS = {m.slug: m for m in [
+    Migrator('domains', [Domain], CouchAttachmentMigrator),
```
Comment (millerdev) [inline corehq/blobs/migrate.py]: We normally write tests for these. See [test_migrate.py](https://github.com/dimagi/commcare-hq/blob/76b69907c78bb26b450c6fb4f3281defc3c1783a/corehq/blobs/tests/test_migrate.py#L164-L201)

**Review by millerdev — COMMENTED** [review]
I'm in the process of implementing a new blob metadata API for the blob db, and doing that at the same time that domains are being migrated might be tricky. What are your time constraints for getting this out?

**Comment by nickpell** [comment]
I don't have any hard time constraints, but it'd be nice to get this out soon to unblock the couchdbkit work.  As an aside, ```CommCareBuild```s is already is in progress: https://github.com/dimagi/commcare-hq/pull/21234

Are your changes deployed and/or being deployed soon?  Would it make sense for me to hurry to get these migrations complete, or to wait?

**Comment by millerdev** [comment]
>  Would it make sense for me to hurry to get these migrations complete, or to wait?

We might both be happy if you can get it done quickly, although I don't want to put unnecessary pressure on you if the migrations are going to be a long process (let's regroup if it looks like it's going to take a long time).

I'm working on an unfortunately monstrous branch to implement the new metadata API, so the changes have not even been reviewed yet, much less deployed.

**Comment by nickpell** [comment]
@millerdev How about I'll keep pushing on the attachment migrations, and when you think you're within 3 days of deploying the new API, I'll stop working on new migrations and focus on wrapping up the existing ones?

**Comment by nickpell** [comment]
Hopefully I'll be done before then

File: corehq/apps/domain/models.py (hunk 1 above)
Comment (nickpell) [reply corehq/apps/domain/models.py]: https://github.com/dimagi/commcare-hq/pull/21239/commits/3d9b87ad7e8996e64aeda48f2864407480841b4c

File: corehq/blobs/migrate.py (hunk 2 above)
Comment (nickpell) [reply corehq/blobs/migrate.py]: https://github.com/dimagi/commcare-hq/pull/21239/commits/24ed7a6f8d65f19e4070e8c4e9c3437fd8b1fa46

**Comment by nickpell** [comment]
✅

**Review by millerdev — APPROVED** [review]

**Pull request merged**